Spanner/src/Connection/Grpc.php

```diff
@@ -1092,4 +1167,71 @@ private function getDatabaseAdminClient()
 
         return $this->databaseAdminClient;
     }
+
+    /**
+     * Allow lazy instantiation of the spanner client.
+     * If routing is enabled, a SpannerClient is returned for a specific instanceName
+     * with a binding in the connection of a specific endpoint.
+     * If the endpoint list is not requested, a request is first made for this instanceName.
+     *
+     * @param string $instanceName Optional. Full name of instance.
+     * @return SpannerClient
+     */
+    private function getSpannerClient($instanceName = null)
+    {
+        if (!$this->spannerClient) {
+            $this->spannerClient = $this->constructGapic(SpannerClient::class, $this->grpcConfig);
+        }
```

Comment:
I think initialization of the default client should be postponed to the case when `enableResourceCaching` is `false`

Reply:
Changed the initialization of `spannerClient` to work by default. Moved this block down so that if `PERMISSION_DENIED` happens in the `enableResourceCaching` section, nevertheless create a client regardless of `enableResourceCaching === false`